fs/backgroundfetcher/background_fetcher_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+   Copyright The Soci Snapshotter Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package backgroundfetcher
+
+import (
+	"compress/gzip"
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/awslabs/soci-snapshotter/cache"
+	spanmanager "github.com/awslabs/soci-snapshotter/fs/span-manager"
+	"github.com/awslabs/soci-snapshotter/soci"
+	"github.com/awslabs/soci-snapshotter/util/testutil"
+	"github.com/opencontainers/go-digest"
+)
+
+func withPauser(p pauser) Option {
+	return func(bf *BackgroundFetcher) error {
+		bf.bfPauser = p
+		return nil
+	}
+}
+
+type countingPauser struct {
+	mu    sync.Mutex
+	count int
+}
+
+func (c *countingPauser) pause(time.Duration) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.count++
+}
+
+func TestBackgroundFetcherPause(t *testing.T) {
+	p := &countingPauser{}
+	bf, err := NewBackgroundFetcher(WithSilencePeriod(0), withPauser(p))
+	if err != nil {
+		t.Fatal(err)
+	}
+	go bf.Run(context.Background())
```

Comment:
How about passing a cancel-able context and cancel that at the end of the function? I'd like to make sure the goroutine is killed at the end of this test.

Reply:
There is a `defer bf.Close()` call. Is there a benefit to having a cancellable context?

Reply:
Oh I wasn't aware about that.